File: service/reservation/reservation-with-events-sink/src/main/java/com/waiter/reservation/domain/Reservation.java
package com.waiter.reservation.domain;

import lombok.Data;
import lombok.RequiredArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;


@Document
@Data
@RequiredArgsConstructor
public class Reservation  {
    @Id
    private Long id;
    private Date createdAt;
    private Date updatedAt;
    private Long storeId;
    private String userName;
    private String reservationDate;
    private Integer userCount;
    private String contactNumber;

    private ReservationStatus reservationStatus;


    public enum ReservationStatus {
        ACCEPT, REJECT, WAIT
    }

    private String message;



    public void updateReservationStatus(ReservationStatus reservationStatus) {
        this.reservationStatus = reservationStatus;
    }

}

File: client/user/src/app/reservation/reservation.ts
import { Store } from "../store/store";

export class Reservation {
    storeId:number;
    userName:string;
    reservationDate:string;
    userCount:number;
    contactNumber:string;
    message:string;

    constructor(store:Store){
        if (store) {
            this.storeId = store.id;
        }
    }
}
